finality-aleph/src/party.rs

```diff
@@ -92,10 +93,20 @@ impl JustificationRequestDelay for JustificationRequestDelayImpl {
     }
 }
 
+impl<B: Block> Verifier<B> for AuthorityVerifier {
+    fn verify(&self, justification: &AlephJustification, hash: B::Hash) -> bool {
+        if !self.is_complete(&hash.encode()[..], &justification.signature) {
+            debug!(target: "afa", "Bad justification for block hash #{:?} {:?}", hash, justification);
```

Comment:
This deserves the `warn` level, as something that shouldn't happen without dishonest nodes.

Reply:
changed